[DOCS] DSB-499: Add docs on securely managing credentials for api-created GX Cloud Data Sources (#8992)
great-expectations/great_expectations: docs/docusaurus/docs/cloud/set_up_gx_cloud.md

    To edit an environment variable, stop the GX Cloud agent, edit the environment variable, save the change, and then restart the GX Cloud agent.

> ## Securely manage credentials for GX API-created Data Sources

Reviewer: ```suggestion
## Securely manage GX API Data Source connection strings 
```

> If you create Data Sources in GX Cloud using the GX API (rather than the GX Cloud web UI), ensure that your sensitive Data Source credentials are properly obfuscated in your connection string. Data Source connection strings are persisted in [GX Cloud backend storage](/docs/cloud/about_gx#gx-cloud-architecture). If you create a connection string containing plaintext credentials, those credentials will be stored in plaintext.

Reviewer: ```suggestion
If you're using the GX API to connect to Data Sources and not GX Cloud, you must obfuscate your sensitive Data Source credentials in your connection string. Data Source connection strings are persisted in [GX Cloud backend storage](/docs/cloud/about_gx#gx-cloud-architecture). Connection strings containing plaintext credentials are stored as plaintext.

The following are additional recommendations for keeping your sensitive data safe: 
```

> 1. Store your credential value in an environment variable. The environment variable name should be prefixed with `GX_CLOUD_`.

Reviewer: ```suggestion
- Store your credential values as environment variables. Prefix environment variable names with `GX_CLOUD_`.
```

> 2. When creating your Data Source connection string, specify the environment variable name in place of where you would normally include the credential. The environment variable name must be encased in curly braces with a preceding dollar sign, for example: `${GX_CLOUD_SNOWFLAKE_PASSWORD}`. Do not include the credential value directly your connection string via interpolation.

Reviewer: ```suggestion
- When creating your Data Source connection string, use an environment variable instead of the credential. Environment variable names must be enclosed by curly braces and be preceded by a dollar sign. For example: `${GX_CLOUD_SNOWFLAKE_PASSWORD}.` Do not use interpolation to add credential values to connection strings.
```

>     A full connection string might look like this:

Reviewer: ```suggestion
    A full connection string appears similar to this example:
```

    ```python title="Example Data Source connection string"
    snowflake://<user-name>:${GX_CLOUD_SNOWFLAKE_PASSWORD}@<account-name>/<database-name>/<schema-name>?warehouse=<warehouse-name>&role=<role-name>
    ```

> 3. Make your credential value available an an environment variable when running the GX Agent.

Reviewer: ```suggestion
- Use environment variables and not credential values when you run the GX Agent. For example:
```

Author: Incorporated via another commit, thanks!

1. Store your credential value as an environment variable by entering `export ENV_VAR_NAME=env_var_value` in the terminal or adding the command to your `~/.bashrc` or `~/.zshrc` file. For example:

    ```bash title="Terminal input"
    export GX_CLOUD_SNOWFLAKE_PASSWORD=<password-string>
    ```
    Prefix environment variable names with `GX_CLOUD_`.

2. Create a Data Source connection string using the environment variable name instead of the credential value. For example:

    ```python title="Example Data Source connection string"
    snowflake://<user-name>:${GX_CLOUD_SNOWFLAKE_PASSWORD}@<account-name>/<database-name>/<schema-name>?warehouse=<warehouse-name>&role=<role-name>
    ```
    Environment variable names must be enclosed by curly braces and be preceded by a dollar sign. For example: `${GX_CLOUD_SNOWFLAKE_PASSWORD}`. Do not use interpolation to add credential values to connection strings.

3. Use the environment variable to supply the credential value when you run the GX Agent. For example:
    ```bash title="Terminal input"

>     docker run --rm -e GX_CLOUD_SNOWFLAKE_PASSWORD="<snowflake_password>" GX_CLOUD_ACCESS_TOKEN="<user_access_token>" -e GX_CLOUD_ORGANIZATION_ID="<organization_id>" -e greatexpectations/agent

Author: ```suggestion
    docker run --rm -e GX_CLOUD_SNOWFLAKE_PASSWORD="<snowflake_password>" -e GX_CLOUD_ACCESS_TOKEN="<user_access_token>" -e GX_CLOUD_ORGANIZATION_ID="<organization_id>" greatexpectations/agent
```

    To edit an environment variable, stop the GX Cloud agent, edit the environment variable, save the change, and then restart the GX Cloud agent.

## Secure your GX API Data Source connection strings

You must obfuscate your sensitive Data Source credentials in your connection string when you use the GX API, not GX Cloud, to create Data Sources. Data Source connection strings are persisted in [GX Cloud backend storage](/docs/cloud/about_gx#gx-cloud-architecture). Connection strings containing plaintext credentials are stored as plaintext.

> 1. Store your credential value as an environment variable. Store the value by entering `export ENV_VAR_NAME=env_var_value` in the terminal or adding the command to your `~/.bashrc` or `~/.zshrc` file. For example:

Reviewer: ```suggestion
1. Store your credential value as an environment variable by entering `export ENV_VAR_NAME=env_var_value` in the terminal or adding the command to your `~/.bashrc` or `~/.zshrc` file. For example:
```

    ```bash title="Terminal input"
    export GX_CLOUD_SNOWFLAKE_PASSWORD=<password-string>
    ```

> 

Reviewer: ```suggestion
```

    Prefix environment variable names with `GX_CLOUD_`.

2. Create a Data Source connection string using the environment variable name instead of the credential value. For example:

    ```python title="Example Data Source connection string"
    snowflake://<user-name>:${GX_CLOUD_SNOWFLAKE_PASSWORD}@<account-name>/<database-name>/<schema-name>?warehouse=<warehouse-name>&role=<role-name>
    ```

> 

Reviewer: ```suggestion
```
